applications/serial_lte_modem/src/main.c

```diff
@@ -338,32 +339,73 @@ static void check_app_fota_status(void)
 int lte_auto_connect(void)
 {
 	int err = 0;
-
-#if defined(CONFIG_SLM_CARRIER)
+#if defined(CONFIG_SLM_AUTO_CONNECT)
 	int stat;
+	struct network_config {
+		/* Network-specific default system mode configuration by %XSYSTEMMODE */
+		int lte_m_support;     /* 0 ~ 1 */
+		int nb_iot_support;    /* 0 ~ 1 */
+		int gnss_support;      /* 0 ~ 1 */
+		int lte_preference;    /* 0 ~ 4 */
+		/* Network-specific default PDN configuration by +CGDCONT and +CGAUTH */
+		bool pdp_config;       /* PDP context definition required or not */
+		char *pdn_fam;         /* PDP type: "Non-IP"; ""IP", "IPV6", "IPV4V6", "Non-IP" */
+		char *pdn_apn;         /* Access point name */
+		char *pdn_username;    /* PDN connection authentication user name */
+		char *pdn_password;    /* PDN connection authentication password */
+		int pdn_auth;          /* Authentication protocol 0(None), 1(PAP), 2(CHAP) */
+	};
+	const struct network_config cfg = {
+#include "slm_net_cfg.h"
+	};
 
-	if (!slm_carrier_auto_connect) {
-		return 0;
-	}
 	err = nrf_modem_at_scanf("AT+CEREG?", "+CEREG: %d", &stat);
 	if (err != 1 || (stat == 1 || stat == 5)) {
 		return 0;
 	}
 
-	LOG_INF("auto connect");
-#if defined(CONFIG_LWM2M_CARRIER_SERVER_BINDING_N)
-	err = nrf_modem_at_printf("AT%%XSYSTEMMODE=0,1,0,0");
-	if (err) {
-		LOG_ERR("Failed to configure RAT: %d", err);
-		return err;
+	LOG_INF("lte auto connect");
+	if (slm_util_casecmp(cfg.pdn_fam, "Non-IP")) {
+		err = nrf_modem_at_printf("AT%%XSYSTEMMODE=0,1,%d,0", cfg.gnss_support);
+		if (err) {
+			LOG_ERR("Failed to configure AcT: %d", err);
+			return err;
+		}
+		err = nrf_modem_at_printf("AT+CGDCONT=0,Non-IP,%s", cfg.pdn_apn);
```

Comment:
Modem could parse it correctly

Reply:
Refer to [sample usage](https://github.com/nrfconnect/sdk-nrf/blob/main/lib/pdn/pdn.c#L446) in NCS.
I think it's OK. For example, instead of `AT+CGDCONT=0,"Non-IP",""`, send `AT+CGDCONT=0,Non-IP,`